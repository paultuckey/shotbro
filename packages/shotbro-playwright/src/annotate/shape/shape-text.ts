import type {TextShape} from "./shape-types";
import type {ShotBroBox} from "../annotate-types";
import {CssPropertiesHyphen, toStyleAttr} from "./css-util";

const defaultProps: TextShape = {
    fontSize: 32,
    color: 'deeppink',
    //padding: 4
    translate: 0,
    translateX: 0,
    translateY: 0,
}

export async function renderText(elPos: ShotBroBox, rawShape: TextShape): Promise<string> {
    const shape = Object.assign({}, defaultProps, rawShape);
    const style = toStyleAttr(toCss(elPos, shape));
    return elPos ? `<div style="${style}">${shape?.value || ''}</div>` : '';
}

function toCss(elPos: ShotBroBox, shape: TextShape): CssPropertiesHyphen {
    const translateX = shape.translateX ?? shape.translate ?? 0
    const translateY = shape.translateY ?? shape.translate ?? 0
    const shadowOffset = (shape?.fontSize || 0) / 16;
    return {
        position: 'fixed',
        top: `${elPos.y + translateX}px`,
        left: `${elPos.x + translateY}px`,
        width: `${elPos.w}px`,
        height: `${elPos.h}px`,
        //padding: `${shape?.padding||0}px`,
        'font-size': `${shape?.fontSize || 0}px`,
        color: shape.color,
        'text-shadow': `${shadowOffset}px ${shadowOffset}px rgba(0,0,0,0.13)`
    };
}


