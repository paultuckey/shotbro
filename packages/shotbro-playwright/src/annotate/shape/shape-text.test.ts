import {renderText} from "./shape-text";
import type {TextShape} from "./shape-types";
import type {ShotBroBox} from "../annotate-types";

describe('Shape Text', () => {

  test('Simple test', async () => {
    const shape: TextShape = {
      value: 'Hello World'
    }
    const elPos: ShotBroBox = {x: 1, y: 2, w: 3, h: 4}
    const html = await renderText(elPos, shape);
    expect(html).toMatchSnapshot('text-simple');
  })
});