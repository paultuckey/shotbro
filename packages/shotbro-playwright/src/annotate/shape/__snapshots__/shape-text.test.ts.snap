// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shape Text Simple test: text-simple 1`] = `"<div style="position:fixed;top:2px;left:1px;width:3px;height:4px;font-size:32px;color:deeppink;text-shadow:2px 2px rgba(0,0,0,0.13);">Hello World</div>"`;
